### PR TITLE
Support for Gradient Bars 

### PR DESCRIPTION
**Pull request by OmarJalil**

### CODE REVIEW

**Comment by misyobun** [comment]
@OmarJalil 
Thank you for your PR. ✨
It looks pretty good.
Let me check.

File: MSBBarChart/Sources/MSBBarChartView.swift [hunk 1]
```diff
@@ -324,6 +383,8 @@ extension MSBBarChartView {
                 isHiddenLabelAboveBar = value
             case let .isHiddenExceptBars(value):
                 isHiddenExceptBars = value
+            case let .isGradientBar(value):
+                isGradientBar = value
```
Comment (misyobun) [inline MSBBarChart/Sources/MSBBarChartView.swift:388]: Variable `isGradientBar` has not been declared yet on MSBBarChartView.

スクリーンショット 2020-06-13 17 35 33

File: MSBBarChart/Sources/MSBBarChartView.swift [hunk 2]
```diff
@@ -153,6 +161,44 @@ extension MSBBarChartView {
...
+
+        mainLayer.addSublayer(barLayer)
+        executeGradientAnimation(xPos, yPos, colors, barLayer)
+    }
+    
+    private func executeGradientAnimation(_ xPos: CGFloat, _ yPos: CGFloat, _ colors: [UIColor], _ layer: CAGradientLayer) {
+        if yPos != mainLayer.frame.size.height - bottomSpace {
+            let barLayerBase = CAGradientLayer()
+            barLayerBase.frame = CGRect(x: xPos, y: self.mainLayer.frame.height - self.bottomSpace, width: self.barWidth, height: 0)
+            barLayerBase.colors = colors
```
Comment (misyobun) [inline MSBBarChart/Sources/MSBBarChartView.swift:182]: Unlike `executeAnimation()`, the `backgroundColor`  of  `CALayer (CAGradientLayer)` is not set in `executeGradientAnimation()`, so the bottom of the bar seems to be rounded.


check


In the case of `executeGradientAnimation()`, the color has a gradient, so
It seems need to create a path with UIBezierPath. 

it's like this.

```
let topRoundPath = UIBezierPath(roundedRect: layer.frame, byRoundingCorners: [.topLeft, .topRight], cornerRadii: CGSize(width: 4.0, height: 4.0))
let mask = CAShapeLayer()
mask.path = topRoundPath.cgPath
layer.mask = mask
```

**Review by misyobun — COMMENTED** [review]
@OmarJalil 
I reviewed.

File: MSBBarChart/Sources/MSBBarChartView.swift (hunk 2 above)
Comment (OmarJalil) [reply MSBBarChart/Sources/MSBBarChartView.swift:182]: Oh okay, i'll check it out!

Comment (OmarJalil) [reply MSBBarChart/Sources/MSBBarChartView.swift:182]: Screen Shot 2020-06-15 at 9 59 29
It disappears when I add the `topRoundPath` :C

**Comment by OmarJalil** [comment]
It is solved and working my friend! 
Screen Shot 2020-06-15 at 10 12 04

File: MSBBarChart/Sources/MSBBarChartView.swift [hunk 3]
```diff
@@ -150,7 +159,47 @@ extension MSBBarChartView {
...
         subAnimation.fillMode = CAMediaTimingFillMode(rawValue: "forwards")
         layer.add(subAnimation, forKey: "bounds")
-        layer.frame = subAnimation.toValue as! CGRect
+        layer.frame = subAnimation.toValue as? CGRect ?? CGRect()
+    }
+
+    private func drawGradientBar(xPos: CGFloat, yPos: CGFloat, colors: [UIColor]) {
+        let barLayer = CAGradientLayer()
+        barLayer.cornerRadius = 4.0
+        barLayer.maskedCorners = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
```
Comment (misyobun) [inline MSBBarChart/Sources/MSBBarChartView.swift:168]: That's a great idea!

File: MSBBarChart/Sources/MSBBarChartView.swift [hunk 4]
```diff
@@ -64,15 +67,16 @@ open class MSBBarChartView: UIView {
...
     
-    private var barLabelValueFontSize:CGFloat = 9.0
+    private var barLabelValueFontSize: CGFloat = 9.0
     
     private var isHiddenLabelAboveBar: Bool = false
     
     private var isHiddenExceptBars: Bool = false
 
-        
+    private var isGradientBar: Bool = false
```
Comment (misyobun) [inline MSBBarChart/Sources/MSBBarChartView.swift:78]: 👍

**Review by misyobun — APPROVED** [review]
@OmarJalil  
It's great work!
Thank you for your contribution.
I'll merge your PR.

File: MSBBarChart/Sources/MSBBarChartView.swift (hunk 3 above)
Comment (OmarJalil) [reply MSBBarChart/Sources/MSBBarChartView.swift:168]: Thank you!

**Pull request merged**